src/crypto/openssl/rsa_public_key.cpp

```diff
@@ -200,6 +200,7 @@ namespace ccf::crypto
     auto hash = OpenSSLHashProvider().Hash(contents, contents_size, md_type);
     Unique_EVP_PKEY_CTX pctx(key);
     CHECK1(EVP_PKEY_verify_init(pctx));
+    CHECK1(EVP_PKEY_CTX_set_rsa_padding(pctx, RSA_PKCS1_PSS_PADDING));
```

Comment:
@achamayou I don't know why, but this's not used in `/app/verifySignature`, however `::sign(` is used in `/app/sign`.

Instead, `bool PublicKey_OpenSSL::verify_hash(`, which is counterintuitive to me, it would've made much more sense to have `sign` `<->` `verify` and `sigh_hash` `<->` `verify_hash`

I though we `sign`/`verify` and `sign_hash

Reply:
Agreed, do we know where verify() is used? verify_hash() should probably be renamed to verify_digest().

Reply:
`do we know where verify() is used`
Ran all e2e tests with extra logs, didn't trigger `verify(` call

Reply:
It looks like it's part of the public C++ API, so we ought to have a unit test for it.

Reply:
UPD. Now it will be used in `js_sign`, and is tested in js e2e logging tests.

And covered with a unit test too.